### ISSUE TITLE
Improve documentation on how to run the project

### ISSUE THREAD

**Issue by ecksun** [ecksun/Contiki-IDS#7]
There seems to be a lack of information on how to run this project, this should be improved.

The best resource at the moment is [./run_test.sh](https://github.com/ecksun/Contiki-IDS/blob/f28f2754504e69c2a3dc8a7cbdf21da0642f79f1/examples/ipv6/ids/evaluation/run_test.sh) which is probably a good starting point as to how to run most of the simulations.

**Comment by philok93** [comment]
Hello,
I want to contribute to the project but I would like to ask you some questions. I have been trying to run the simulations and I managed to run it with cooja and not with run_test.sh. I want to understand how attacks are triggered and then check the IDS performance to see if I can get the results of your paper.
My questions are:
1) is there any difference between ContikiIDS-evil/.../udp_sender.c and ContikiIDS/.../udp_sender.c? Because I am trying to understand how the sinkhole/clone/selective forwarding attacks are triggered so that IDS will detect them. Even if I tried cloning the master branch and running ids_demo_simulation, I didn't manage to trigger alarm and have an inconsistency in a node. How can I achieve that? 
2) Could you please tell me which files should I check to find the implementation of different attacks? I tried to find the implementation of attacks by cloning "attacks" branch but I didn't find anything about it.
3) I try to understand the logic behind IDS but I need to know, is the message "nodes with outdated information" means that is a possible attack node or it's a normal message? 

I will try to make SVELTE work in Contiki 3 but I face many problems. I would really appreciate if you could answer some of my questions to improve this project.

Thank you!

**Comment by zhangedgar** [comment]
Hi@philok93 ,
I have same questions with you, can't find the attacks' implement.
I'm trying to implement attacks in term of the theories.
If you have new findings, please let me know. E-mail: cybersec.zhang@gmail.com

Thanks!

**Comment by ecksun** [comment]
> is there any difference between ContikiIDS-evil/.../udp_sender.c and ContikiIDS/.../udp_sender.c? Because I am trying to understand how the sinkhole/clone/selective forwarding attacks are triggered so that IDS will detect them. Even if I tried cloning the master branch and running ids_demo_simulation, I didn't manage to trigger alarm and have an inconsistency in a node. How can I achieve that?

I'm not entirely sure which udp-senders you are referring to, may I suggest just diffing them?

> Could you please tell me which files should I check to find the implementation of different attacks? I tried to find the implementation of attacks by cloning "attacks" branch but I didn't find anything about it.

I seem to have implemented the different attacks on different branches, try checking out the changes made to the different branches.

> I try to understand the logic behind IDS but I need to know, is the message "nodes with outdated information" means that is a possible attack node or it's a normal message?

From the code and the docs it seems like it indicates an issue, especially this line: `if (!valid_node(&network[i]))`

**Comment by chanseng** [comment]
hello
I seem not to find the folder ContikiIDS-evil,please help me.:-)

**Comment by AndreAlves1996** [comment]
> 1. is there any difference between ContikiIDS-evil/.../udp_sender.c and ContikiIDS/.../udp_sender.c?

Hi @philok93,

Did you manage to find out which kind or firmware difference is there between the required "evil-motes" by Cooja when we try to build a simulation form the project?

**Comment by SalalAmjad** [comment]
> There seems to be a lack of information on how to run this project, this should be improved.
> 
> The best resource at the moment is [./run_test.sh](https://github.com/ecksun/Contiki-IDS/blob/f28f2754504e69c2a3dc8a7cbdf21da0642f79f1/examples/ipv6/ids/evaluation/run_test.sh) which is probably a good starting point as to how to run most of the simulations.

Hi @ecksun,
Can you please tell me how to run simulations using **run_test.sh** file. How to run Contiki-IDS simulations in cooja? We downloaded this software but don't know how to use it. Please guide me.

**Comment by ecksun** [comment]
> Can you please tell me how to run simulations using run_test.sh file

No, I can't tell you more than what is in the script, I'm sorry.

I would recommend you read it, its not that long. I see that it isn't perfect (unquoted variables for example) but the gist of it should still be quite easy to grasp.
